src/main/java/net/minestom/server/tab/TabList.java

```diff
@@ -0,0 +1,304 @@
+     * the gamemode of the specified player as the specified gamemode
+     * <p>
+     * This is used to communicate part of a player's actual gamemode so it's recommended you only use this method with a {@link net.minestom.server.entity.fakeplayer.FakePlayer}
+     *
+     * @param player   The player to set the fake gamemode for
+     * @param gameMode The gamemode to send for the specified player
+     * @throws IllegalStateException if the player is not on the TabList or the TabList is set to automatically update gamemodes
+     */
+    public void setDisplayedGamemode(@NotNull Player player, @NotNull GameMode gameMode) {
+        if (this.gamemodeUpdates)
+            throw new IllegalStateException("Cannot set fake gamemode unless gamemodeUpdates is set to false");
+        if (!this.displayedPlayers.contains(player))
+            throw new IllegalStateException("Cannot set fake gamemode for a player not displayed on the TabList");
+        PlayerInfoPacket playerInfoPacket = new PlayerInfoPacket(PlayerInfoPacket.Action.UPDATE_GAMEMODE);
+        playerInfoPacket.playerInfos.add(new PlayerInfoPacket.UpdateGamemode(player.getUuid(), gameMode));
+
+        PacketUtils.sendGroupedPacket(this.viewers, playerInfoPacket);
+    }
+
+    /**
+     * Sends a packet to all viewers of the TabList telling them to display
+     * the latency of the specified player as the specified latency
+     *
+     * @param player  The player to set the fake latency for
+     * @param latency The latency to send for the specified player
+     * @throws IllegalStateException if the player is not on the TabList or the TabList is set to automatically update latency
+     */
+    public void setDisplayedPing(@NotNull Player player, int latency) {
+        if (this.latencyUpdates)
+            throw new IllegalStateException("Cannot set fake latency unless latencyUpdates is set to false");
+        if (!this.displayedPlayers.contains(player))
+            throw new IllegalStateException("Cannot set fake latency for a player not displayed on the TabList");
+
+        PlayerInfoPacket playerInfoPacket = new PlayerInfoPacket(PlayerInfoPacket.Action.UPDATE_LATENCY);
+        playerInfoPacket.playerInfos.add(new PlayerInfoPacket.UpdateLatency(player.getUuid(), latency));
+
+        PacketUtils.sendGroupedPacket(this.viewers, playerInfoPacket);
+    }
+
+    private PlayerInfoPacket.AddPlayer.Property createPropertyPacket(Player player) {
+        final String textures = player.getSkin().getTextures();
+        final String signature = player.getSkin().getSignature();
+
+        return new PlayerInfoPacket.AddPlayer.Property("textures", textures, signature);
+    }
+
+    /**
+     * Whether the TabList is configured to automatically send player latency
+     * updates to all its viewers
+     *
+     * @return boolean whether player latency will be auto-updated on TabLists
+     */
+    public boolean isLatencyUpdates() {
```

Comment:
I wonder if this could be named better - maybe `hasLatencyUpdates()`?

Reply:
It isn't has since it's not a queue/updating system it's a toggle as to whether the tablist will automatically update latency, or if latency is spoofed.

Reply:
I still think `isLatencyUpdates` just doesn't make any sense. Perhaps `doesLatencyUpdates` or `handlesLatencyUpdates`?

Reply:
I agree will change to doesLatencyUpdates